srml/balances/src/tests.rs

```diff
@@ -21,206 +21,328 @@
+				"account liquidity restrictions prevent withdrawal"
+			);
+		});
+}
+
+#[test]
+fn partial_locking_should_work() {
+	ExtBuilder::default()
+		.existential_deposit(1)
+		.monied(true)
+		.build()
+		.execute_with(|| {
+			Balances::set_lock(
+				ID_1, 
+				&1, 
+				WithdrawLock::Normal(NormalLock {
+					amount: 5, 
+					until: u64::max_value(),
+				}), 
+				WithdrawReasons::all()
+			);
+			assert_ok!(<Balances as Currency<_>>::transfer(&1, &2, 1, AllowDeath));
+		});
+}
+
+#[test]
+fn lock_removal_should_work() {
+	ExtBuilder::default()
+		.existential_deposit(1)
+		.monied(true)
+		.build()
+		.execute_with(|| {
+			Balances::set_lock(
+				ID_1, 
+				&1, 
+				WithdrawLock::Normal(NormalLock {
+					amount: u64::max_value(), 
+					until: u64::max_value(),
+				}), 
+				WithdrawReasons::all()
+			);
+			Balances::remove_lock(ID_1, &1);
+			assert_ok!(<Balances as Currency<_>>::transfer(&1, &2, 1, AllowDeath));
+		});
+}
+
+#[test]
+fn lock_replacement_should_work() {
+	ExtBuilder::default()
+		.existential_deposit(1)
+		.monied(true)
+		.build()
+		.execute_with(|| {
+			Balances::set_lock(
+				ID_1, 
+				&1,
+				WithdrawLock::Normal(NormalLock {
+					amount: u64::max_value(), 
+					until: u64::max_value(),
+				}), 
+				WithdrawReasons::all()
+			);
+			Balances::set_lock(
+				ID_1, 
+				&1, 
+				WithdrawLock::Normal(NormalLock {
+					amount: 5, 
+					until: u64::max_value(),
+				}),
+				WithdrawReasons::all()
+			);
+			assert_ok!(<Balances as Currency<_>>::transfer(&1, &2, 1, AllowDeath));
+		});
+}
+
+#[test]
+fn double_locking_should_work() {
+	ExtBuilder::default()
+		.existential_deposit(1)
+		.monied(true)
+		.build()
+		.execute_with(|| {
+			Balances::set_lock(
+				ID_1, 
+				&1, 
+				WithdrawLock::Normal(NormalLock {
+					amount: 5, 
+					until: u64::max_value(),
+				}),
+				WithdrawReasons::all()
+			);
+			Balances::set_lock(
+				ID_2, 
+				&1, 
+				WithdrawLock::Normal(NormalLock {
+					amount: 5, 
+					until: u64::max_value(),
+				}),
+				WithdrawReasons::all()
+			);
+			assert_ok!(<Balances as Currency<_>>::transfer(&1, &2, 1, AllowDeath));
+		});
+}
+
+#[test]
+fn combination_locking_should_work() {
+	ExtBuilder::default()
+		.existential_deposit(1)
+		.monied(true)
+		.build()
+		.execute_with(|| {
+			Balances::set_lock(
+				ID_1, 
+				&1, 
+				WithdrawLock::Normal(NormalLock {
+					amount: u64::max_value(), 
+					until: 0,
+				}),
+				WithdrawReasons::all()
+			);
+			Balances::set_lock(
+				ID_2, 
+				&1, 
+				WithdrawLock::Normal(NormalLock {
+					amount: 0, 
+					until: u64::max_value(),
+				}),
+				WithdrawReasons::all()
+			);
+			Balances::set_lock(
+				ID_3, 
+				&1, 
+				WithdrawLock::Normal(NormalLock {
+					amount: 0, 
+					until: 0,
+				}),
+				WithdrawReasons::all()
+			);
+			assert_ok!(<Balances as Currency<_>>::transfer(&1, &2, 1, AllowDeath));
+		});
+}
+
+#[test]
+fn lock_value_extension_should_work() {
```

Comment:
currently we shadow the `LockableCurrency::extend_lock()`

please add TODO to this test case